src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -0,0 +1,332 @@
+            Optional<Switch> coupler = propagatedContingency.switchesToOpen.stream().filter(PropagatedContingency::isCoupler).findFirst();
+            if (coupler.isEmpty()) {
+                propagatedContingencies.add(propagatedContingency);
+            } else {
+                // Sensitivity analysis works in bus view, it cannot deal (yet)  with contingencies whose propagation encounters a coupler
+                LOGGER.warn("Propagated contingency '{}' not processed: coupler '{}' has been encountered while propagating the contingency",
+                    contingency.getId(), coupler.get().getId());
+            }
+        }
+        return propagatedContingencies;
+    }
+
+    public static List<PropagatedContingency> createListForSecurityAnalysis(Network network, List<Contingency> contingencies,
+                                                                            Set<Switch> allSwitchesToOpen, boolean shuntCompensatorVoltageControlOn,
+                                                                            boolean slackDistributionOnConformLoad) {
+        List<PropagatedContingency> propagatedContingencies = new ArrayList<>();
+        for (int index = 0; index < contingencies.size(); index++) {
+            Contingency contingency = contingencies.get(index);
+            PropagatedContingency propagatedContingency =
+                    PropagatedContingency.create(network, contingency, index, shuntCompensatorVoltageControlOn, true, slackDistributionOnConformLoad);
+            propagatedContingencies.add(propagatedContingency);
+            allSwitchesToOpen.addAll(propagatedContingency.switchesToOpen);
+        }
+        return propagatedContingencies;
+    }
+
+    private static PropagatedContingency create(Network network, Contingency contingency, int index, boolean shuntCompensatorVoltageControlOn,
+                                                boolean withBreakers, boolean slackDistributionOnConformLoad) {
+        Set<Switch> switchesToOpen = new HashSet<>();
+        Set<Terminal> terminalsToDisconnect =  new HashSet<>();
+        Set<String> branchIdsToOpen = new HashSet<>();
+        Set<String> hvdcIdsToOpen = new HashSet<>();
+        Set<Load> loadsToLose = new HashSet<>();
+        Set<Generator> generatorsToLose = new HashSet<>();
+        Set<ShuntCompensator> shuntsToLose = new HashSet<>();
+
+        // process elements of the contingency
+        for (ContingencyElement element : contingency.getElements()) {
+            switch (element.getType()) {
+                case BRANCH:
+                case LINE:
+                case TWO_WINDINGS_TRANSFORMER:
+                    // branch check is done inside tripping
+                    ContingencyTripping.createBranchTripping(network, element.getId())
+                        .traverse(switchesToOpen, terminalsToDisconnect);
```

Comment:
I removed here the adding to branchIdsToOpen, because it seems to me redundant with the post propagation processing as terminalsToDisconnect contains to initial starting equipment. Seems ok to you?

Reply:
Indeed I really don't know because it was my question to Florian but it tells me that it is not from him.